pyHalo/PresetModels/external.py

```diff
@@ -52,107 +70,106 @@ def DMFromGalacticus(z_lens,z_source,galacticus_hdf5,tree_index, kwargs_cdm,mass
 
     params_to_read = galacticus_params_additional + PARAMS_TO_READ_DEF
     
-    # we create a realization of only line-of-sight halos by setting sigma_sub = 0.0
-    # only include these halos if requested
-    kwargs_cdm['sigma_sub'] = 0.0
+    if isinstance(galacticus_hdf5,str):
+        nodedata = gutil.read_nodedata_galacticus(galacticus_hdf5,params_to_read=params_to_read)
+    elif isinstance(galacticus_hdf5, h5py.Group):
+        nodedata = gutil.hdf5_read_galacticus_nodedata(galacticus_hdf5,params_to_read=params_to_read)
+    else:
+        nodedata = galacticus_hdf5
+
+    z_lens = np.mean(nodedata[gutil.PARAM_Z_LAST_ISOLATED][np.logical_not(nodedata_filter_subhalos(nodedata,gutil))])
 
-    los_norm = 1 if include_field_halos else 0
-    los_norm = kwargs_cdm.get("LOS_normalization") if not kwargs_cdm.get("LOS_normalization") is None else los_norm
+    if tree_index is None:
+        tree_index = np.random.choice(np.unique(nodedata[gutil.PARAM_TREE_INDEX]))
```

Comment:
I assume tree_index should be an integer? I would add an else clause here that reads 

else: 
    tree_index = int(np.round(tree_index))
    
so that you have the option of sampling tree_index from a distribution